### PR TITLE
Improve ux

### PR DESCRIPTION
**Pull request by gemcfadyen**
The grid display has been updated to be one based.

This means that when the HumanPlayer enters a valid choice (number between 1-9 inclusive), the number is passed straight onto the board class.  I have tried to keep the offset amendments (+/- 1) in one place (Board.java) so that if the grid representation should change, only one class would be effected.

The class Cell has also been introduced to encapsulate the offset and the symbol of each position of the grid. Additionally to allow the prompt to display the board, a  new method 'getRows()' has been added to Board.java.

A separate pull request will be raised for clearing the screen and redrawing the grid each time a player takes a turn.

### CODE REVIEW

**Comment by gemcfadyen** [comment]
@jsuchy @ecomba - If you would like to have a look at this change then feel free, if not I'll merge it before Thu's IPM and take you through it then.

**Comment by ecomba** [comment]
This is unrelated to the current task, but something that I stumbled upon while trying to run your code locally.

Can you specify the dependencies of your project somewhere? When cloning from a repository it would be nice if you were able to compile and run the tests following a "common" approach (or at least have the steps to be able to run the tests and the program itself in the README).

The problem I was facing right now was that while trying to run the tests IDEA erred with the following:

```
/Users/ecomba/development/8thlight/apprenticeship/georgina/Apprenticeship-JavaTicTacToe/src/test/java/ttt/WriteToPromptExceptionTest.java

Error:(29, 9) java: no suitable method found for assertThat(java.lang.Throwable,org.hamcrest.Matcher<java.io.IOException>)
    method org.junit.Assert.<T>assertThat(java.lang.String,T,org.hamcrest.Matcher<? super T>) is not applicable
      (cannot infer type-variable(s) T
        (actual and formal argument lists differ in length))
    method org.junit.Assert.<T>assertThat(T,org.hamcrest.Matcher<? super T>) is not applicable
      (inferred type does not conform to upper bound(s)
        inferred: java.lang.Throwable
        upper bound(s): java.io.IOException,java.lang.Object)
```

```
/Users/ecomba/development/8thlight/apprenticeship/georgina/Apprenticeship-JavaTicTacToe/src/test/java/ttt/ReadFromPromptExceptionTest.java

Error:(28, 9) java: no suitable method found for assertThat(java.lang.Throwable,org.hamcrest.Matcher<java.io.IOException>)
    method org.junit.Assert.<T>assertThat(java.lang.String,T,org.hamcrest.Matcher<? super T>) is not applicable
      (cannot infer type-variable(s) T
        (actual and formal argument lists differ in length))
    method org.junit.Assert.<T>assertThat(T,org.hamcrest.Matcher<? super T>) is not applicable
      (inferred type does not conform to upper bound(s)
        inferred: java.lang.Throwable
        upper bound(s): java.io.IOException,java.lang.Object)
```

Somehow the `hamcrest` matcher is not the one it should be, which leads me to believe it is a versioning problem.

Can you provide the versions you have been using?

File: src/test/java/ttt/BoardTest.java [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 import org.junit.Test;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
```
Comment (ecomba) [inline src/test/java/ttt/BoardTest.java]:
```
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
```

These classes are not used and can be removed.

_If there are unused imports you can always optimise them using a shortcut `control + option + o`_

File: src/test/java/ttt/CellTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package ttt;
+
+import org.hamcrest.Matchers;
```
Comment (ecomba) [inline src/test/java/ttt/CellTest.java]: Unused class Matchers

File: src/test/java/ttt/CellTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.*;
```
Comment (ecomba) [inline src/test/java/ttt/CellTest.java]: `import org.hamcrest.Matchers` and `import static org.hamcrest.Matchers.*` can be removed as they are not being used.

File: src/test/java/ttt/PromptSpy.java [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 import com.sun.jdi.connect.Connector;
 
+import java.util.List;
+
 import static ttt.Board.*;
```
Comment (ecomba) [inline src/test/java/ttt/PromptSpy.java]: `import com.sun.jdi.connect.Connector`, `import java.util.List` and `import static ttt.Board.*` are not being used.

File: src/test/java/ttt/PromptSpy.java [hunk 5]
```diff
@@ -48,9 +50,14 @@ public int getNumberOfTimesDrawMessageHasBeenPrinted() {
...
-        for (int i = 0; i < BOARD_DIMENSION * BOARD_DIMENSION; i++) {
-            gridFormation.append(lastBoardPrinted.getSymbolAt(i));
+        Cell[][] rows = lastBoardPrinted.getRows();
+
+        for(Cell[] cells: rows) {
+            for(int i = 0; i < cells.length; i++) {
+                gridFormation.append(lastBoardPrinted.getSymbolAt(cells[i].getOffset()));
+            }
         }
+
```
Comment (ecomba) [inline src/test/java/ttt/PromptSpy.java:54]: Have you considered using foreach here?

```
for (Cell cell : cells) {
    gridFormation.append(lastBoardPrinted.getSymbolAt(cell.getOffset()));
}
```

IntelliJ does this with one `option+enter` command.

**Comment by gemcfadyen** [comment]
Thanks for the comments @ecomba - I have updated the PR. Regarding the hamcrest version - I am using the version that is shipped within JUnit4.11 which is on my classpath (which contains hamcrest-core-1.3.jar, hamcrest-library-1.3.jar) - I have not added any external dependencies explicitly.

**Pull request merged**